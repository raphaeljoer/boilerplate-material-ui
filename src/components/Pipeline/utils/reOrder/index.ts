import { DropResult } from 'react-beautiful-dnd';
import { PipelineProps } from '../../types';

export const index = (
  pipeline: PipelineProps,
  result: DropResult
): PipelineProps | undefined => {
  const { destination, source, draggableId } = result;

  if (!destination) return;
  if (
    destination.droppableId === source.droppableId &&
    destination.index === source.index
  )
    return;

  const startColumn = pipeline.columns[source.droppableId];
  const endColumn = pipeline.columns[destination.droppableId];

  if (startColumn === endColumn) {
    //reorder within the same column
    const newItemsId = Array.from(startColumn.itemIds);
    newItemsId.splice(source.index, 1);
    newItemsId.splice(destination.index, 0, draggableId);

    const newColumn = { ...startColumn, itemIds: newItemsId };
    const newState = {
      ...pipeline,
      columns: {
        ...pipeline.columns,
        [newColumn.id]: newColumn
      }
    };

    return newState;
  }

  //reorder between different columns
  const newItemsId = Array.from(startColumn.itemIds);
  newItemsId.splice(source.index, 1);

  const newStartColumn = { ...startColumn, itemIds: newItemsId };

  const newEndItemsId = Array.from(endColumn.itemIds);
  newEndItemsId.splice(destination.index, 0, draggableId);
  const newEndColumn = { ...endColumn, itemIds: newEndItemsId };

  const newState = {
    ...pipeline,
    columns: {
      ...pipeline.columns,
      [newStartColumn.id]: newStartColumn,
      [newEndColumn.id]: newEndColumn
    }
  };

  return newState;
};
