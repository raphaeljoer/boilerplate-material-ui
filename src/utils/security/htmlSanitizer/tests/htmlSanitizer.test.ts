import { render } from 'config/tests/utils';
import { htmlSanitizer } from '..';

describe('htmlSanitizer', () => {
  test('Should escape <alert> malicious code', () => {
    const html = '<script>alert("xss")</script>';
    const { container } = render(htmlSanitizer(html));
    expect(container).toMatchSnapshot();
    expect(container).toMatchInlineSnapshot(
      `
      <div>
        <div>
          &lt;script&gt;alert("xss")&lt;/script&gt;
        </div>
      </div>
      `
    );
  });
  test('Should not escape <a>', () => {
    const html = '<a href="https://google.com">google</a>';
    const { container } = render(htmlSanitizer(html));
    expect(container).toMatchSnapshot();
  });
});
