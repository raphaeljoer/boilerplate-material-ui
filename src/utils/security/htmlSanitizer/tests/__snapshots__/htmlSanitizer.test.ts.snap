// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`htmlSanitizer Should escape <alert> malicious code 1`] = `
<div>
  <div>
    &lt;script&gt;alert("xss")&lt;/script&gt;
  </div>
</div>
`;

exports[`htmlSanitizer Should not escape <a> 1`] = `
<div>
  <div>
    <a
      href="https://google.com"
    >
      google
    </a>
  </div>
</div>
`;
